src/main/java/aeneas/controllers/ChangeLevelTypeMove.java

```diff
@@ -1,50 +1,69 @@
 package aeneas.controllers;
 import aeneas.models.Level;
-import aeneas.models.Model;
+import aeneas.models.Level.LevelType;
+import aeneas.models.LightningLevel;
+import aeneas.models.PuzzleLevel;
+import aeneas.models.ReleaseLevel;
+import aeneas.views.BuildLevelView;
 
 /**
  * Move action to change the type of a level
- * @author Logan
- *
  */
 public class ChangeLevelTypeMove implements IMove {
 
+  Level oldLevel;
 
-  Level level;
+  LevelType type;
 
-  Model model;
-
-  int type;
+  BuildLevelView view;
 
   /**
    * Constructor
    *
    * @param level The level that is currently being played
-   * @param model The game model that holds the level being changed
+   * @param view The build level view.
    * @param type the type of the level to be changed to
    */
-  public ChangeLevelTypeMove(Level level, Model model, int type) {
-    this.level = level;
-    this.model = model;
+  public ChangeLevelTypeMove(Level level, BuildLevelView view, LevelType type) {
+    this.oldLevel = level;
     this.type = type;
+    this.view = view;
   }
 
   @Override
   public boolean execute() {
-    // TODO Auto-generated method stub
-    return false;
+    if (!isValid()) return false;
+
+    Level newLevel = null;
+    switch (type) {
+      case PUZZLE:
+        newLevel = new PuzzleLevel(oldLevel);
+        break;
+      case LIGHTNING:
+        newLevel = new LightningLevel(oldLevel);
+        break;
+      case RELEASE:
+        newLevel = new ReleaseLevel(oldLevel);
+        break;
+    }
```

Comment:
- Remove the FXML that makes the three options.
 - Add a list of level types in the ``Model```
 - Each level defines getter's for it's Icon, english name

There ought to be a way to avoid any kind of string comparison/lookup